mods/taskbar-empty-space-clicks.wh.cpp

```diff
@@ -0,0 +1,1572 @@
+// ==WindhawkMod==
+// @id              taskbar-empty-space-clicks
+// @name            Click on empty taskbar space
+// @description     Trigger custom action when empty space on a taskbar is double/middle clicked.
+// @version         1.0
+// @author          m1lhaus
+// @github          https://github.com/m1lhaus
+// @include         explorer.exe
+// @compilerOptions -DWINVER=0x0602 -D_WIN32_WINNT=0x0602 -lcomctl32 -loleaut32 -lole32 -lversion
+// ==/WindhawkMod==
+
+// Source code is published under The GNU General Public License v3.0.
+//
+// For bug reports and feature requests, please open an issue here:
+// https://github.com/m1lhaus/windhawk-mods/issues
+//
+// For pull requests, development takes place here:
+// https://github.com/m1lhaus/windhawk-mods
+
+// ==WindhawkModReadme==
+/*
+# Click on empty taskbar space
+
+This mod lets you assign an action to a mouse click on Windows taskbar. Double-click and middle-click actions are supported. This mod only modifies behaviour when empty space of the taskbar is clicked. Buttons, menus or other function of the taskbar are not affected. Both primary and secondary taskbars are supported.
+
+## Supported actions:
+
+1. **Show desktop** - Toggle show/hide desktop
+2. **Ctrl+Alt+Tab** - Opens Ctrl+Alt+Tab dialog
+3. **Task Manager** - Opens Windows default task manager
+4. **Mute system volume** - Toggle mute of system volume (all sound)
+5. **Taskbar auto-hide** - Toggle Windows taskbar auto-hide feature 
+6. **Win+Tab** - Opens Win+Tab dialog
+7. **Hide desktop icons** - Toggle show/hide of all desktop icons
+
+## Example
+
+Following animation shows **Taskbar auto-hide** feature. Feature gets toggled whenever user double-clicks the empty space on a taskbar.
+
+![Demonstration of Toggle taskbar autohide mod for Windhawk](https://i.imgur.com/BRQrVnX.gif)
+
+## Supported Windows versions are:
+- Windows 10 22H2 (prior versions are not tested, but should work as well)
+- Windows 11 21H2 - latest major
+
+I will not supporting Insider preview or other minor versions of Windows. However, feel free to [report any issues](https://github.com/m1lhaus/windhawk-mods/issues) related to those versions. I'll appreciate the heads-up in advance.  
+
+In case you are using old Windows taskbar on Windows 11 (**Explorer Patcher** or a similar tool), enable corresponding option on Settings menu. This options will be tested only with the latest major version of Windows 11 (e.g. 23H2). 
+
+*/
+// ==/WindhawkModReadme==
+
+// ==WindhawkModSettings==
+/*
+- doubleClickAction: nothing
```

Comment:
Replace `nothing` with `ACTION_NOTHING`, otherwise the default text just displays "nothing".

![image](https://github.com/ramensoftware/windhawk-mods/assets/4129781/cd80f763-783e-4aaf-80be-4d9de86c6846)